libosmscout-client/src/osmscoutclient/DBThread.cpp

```diff
@@ -508,18 +508,20 @@ CancelableFuture<bool> DBThread::FlushCaches(const std::chrono::milliseconds &id
 
 void DBThread::RunJob(AsynchronousDBJob job)
 {
-  std::shared_lock locker(lock);
+  ReadLock locker(latch);
   if (!isInitializedInternal()){
     locker.unlock();
     osmscout::log.Warn() << "ignore request, dbs is not initialized";
     return;
   }
+  // delegates the lock state to task
+  locker.unlock();
   job(basemapDatabase,databases,std::move(locker));
```

Comment:
This is not correct, when you are not holding read-lock, list of database may be changed!

Reply:
Here the call of job() is synchronous, so I can call try_lock_share() near and no longer delegate the lock to the task. To be tested.